changes_revision.go

```diff
@@ -140,6 +141,35 @@ type PatchOptions struct {
 	Path string `url:"path,omitempty"`
 }
 
+// StringPointerWriter is a new type based on *string.
+// We would like it to implement Writer interface so that we can avoid
+// unmarshalling of non JSON responses by Client.Do
+type StringPointerWriter struct {
+	Target *string
+}
+
+func NewStringPointerWriter(target *string) (*StringPointerWriter, error) {
+	if target == nil {
+		return nil, errors.New("StringPointerWriter: target *string cannot be nil")
+	}
+	return &StringPointerWriter{Target: target}, nil
+}
+
+// Write implements the io.Writer interface for *StringPointerWriter.
+func (spw *StringPointerWriter) Write(p []byte) (n int, err error) {
+	// Check if the StringPointerWriter pointer itself is nil, or if its Target is nil.
+	if spw == nil || spw.Target == nil {
+		return 0, errors.New("StringPointerWriter: receiver or target *string is nil, cannot write")
+	}
+
+	// *(spw.Target) gives us the actual string value.
+	// We append the new data (converted from []byte to string) to it.
+	*(spw.Target) = *(spw.Target) + string(p)
```

Comment:
Appending a string to a string causes a copy and and allocation, so when you have a need to append, it's generally better to stick with []byte which can append without copying and allocating whenever there's enough capacity in the slice.